waku/waku_rln_relay/rln_relay.nim

```diff
@@ -89,13 +89,29 @@ type WakuRLNRelay* = ref object of RootObj
   groupManager*: GroupManager
   onFatalErrorAction*: OnFatalErrorHandler
   nonceManager*: NonceManager
+  epochMonitorFuture*: Future[void]
 
 proc calcEpoch*(rlnPeer: WakuRLNRelay, t: float64): Epoch =
   ## gets time `t` as `flaot64` with subseconds resolution in the fractional part
   ## and returns its corresponding rln `Epoch` value
   let e = uint64(t / rlnPeer.rlnEpochSizeSec.float64)
   return toEpoch(e)
 
+proc nextEpochTime*(rlnPeer: WakuRLNRelay, t: float64): float64 =
+  var leftPtr = t
+  var rightPtr = t + rlnPeer.rlnEpochSizeSec.float64
```

Comment:
I think we should convert both pointers to uint64 to be more precise (as we calculate epochs based on uints) and to avoid bugs related to both pointers not converging properly because of floating point differences.

Apart from that, logic looks good but I would make sure sure to test it well as I'm not smart enough to be 100% sure about it at first sight lol.

Otherwise, we can just brute force it instead of doing this kind of binary search, as it might be easier to follow, less bug prone, and at the end of the day it's just a loop with max 10 iterations which is nothing :)